benchmarks/src/main/java/zipkin2/it/ServerIntegratedBenchmark.java

```diff
@@ -0,0 +1,302 @@
+
+  Closer closer;
+
+  @BeforeEach void setUp() {
+    closer = Closer.create();
+  }
+
+  @AfterEach void tearDown() throws Exception {
+    closer.close();
+  }
+
+  @Test void inMemory() throws Exception {
+    runBenchmark(null);
+  }
+
+  @Test void elasticsearch() throws Exception {
+    GenericContainer<?> elasticsearch = new GenericContainer<>("openzipkin/zipkin-elasticsearch7")
+      .withNetwork(Network.SHARED)
+      .withNetworkAliases("elasticsearch")
+      .withLabel("name", "elasticsearch")
+      .withExposedPorts(9200)
+      .waitingFor(new HttpWaitStrategy().forPath("/_cluster/health"));
+    startContainer(elasticsearch);
+
+    runBenchmark(elasticsearch);
+  }
+
+  @Test void cassandra() throws Exception {
+    GenericContainer<?> cassandra = new GenericContainer<>("openzipkin/zipkin-cassandra")
+      .withNetwork(Network.SHARED)
+      .withNetworkAliases("cassandra3")
+      .withLabel("name", "cassandra3")
+      .withExposedPorts(9042)
+      .waitingFor(new LogMessageWaitStrategy().withRegEx(".*Starting listening for CQL clients.*"));
+    startContainer(cassandra);
+
+    runBenchmark(cassandra);
+  }
+
+  @Test void mysql() throws Exception {
+    GenericContainer<?> mysql = new GenericContainer<>("openzipkin/zipkin-mysql")
+      .withNetwork(Network.SHARED)
+      .withNetworkAliases("mysql")
+      .withLabel("name", "mysql")
+      .withExposedPorts(3306);
+    startContainer(mysql);
+
+    runBenchmark(mysql);
+  }
+
+  void runBenchmark(@Nullable GenericContainer<?> storage) throws Exception {
+    GenericContainer<?> zipkin = startZipkin(storage);
+
+    GenericContainer<?> backend = new GenericContainer<>("openzipkin/example-sleuth-webmvc")
+      .withNetwork(Network.SHARED)
+      .withNetworkAliases("backend")
+      .withCommand("backend")
+      .withExposedPorts(9000)
+      .waitingFor(new HttpWaitStrategy().forPath("/actuator/health"));
+
+    GenericContainer<?> frontend = new GenericContainer<>("openzipkin/example-sleuth-webmvc")
+      .withNetwork(Network.SHARED)
+      .withNetworkAliases("frontend")
+      .withCommand("frontend")
+      .withExposedPorts(8081)
+      .waitingFor(new HttpWaitStrategy().forPath("/actuator/health"));
+
+    if (RELEASED_ZIPKIN_VERSION == null) {
+      Testcontainers.exposeHostPorts(9411);
+      String zipkinArg = "-Dspring.zipkin.baseUrl=http://host.testcontainers.internal:9411";
+      backend.withEnv("JAVA_OPTS", zipkinArg);
+      frontend.withEnv("JAVA_OPTS", zipkinArg);
+    }
+
+    startContainer(backend);
+    startContainer(frontend);
+
+    String prometheusResource = RELEASED_ZIPKIN_VERSION == null
+      ? "prometheus-local.yml" : "prometheus-container.yml";
+    GenericContainer<?> prometheus = new GenericContainer<>("prom/prometheus")
+      .withNetwork(Network.SHARED)
+      .withNetworkAliases("prometheus")
+      .withExposedPorts(9090)
+      .withCopyFileToContainer(
```

Comment:
`withClasspathResourceMapping` should work too

Reply:
Yeah I initially used that but had an error like "already mounted" or something once on Windows. Since these files are static copying seems less tricky than binding.

Reply:
Ouch.... Actually, I just recall that I submitted https://github.com/testcontainers/testcontainers-java/issues/1163 but we never did the change, but we should :)

Yes, copy is better here :) Meanwhile, I will make sure that https://github.com/testcontainers/testcontainers-java/issues/1163 is fixed soon :)